CHANGELOG.rst

```diff
@@ -17,6 +17,14 @@ Unreleased
 ----------
 None
 
+[3.46.6]
+--------
+fix: Return None for context if a role assignment has no enterprise_customer and does not apply to all contexts
```

Comment:
Nit: This is actually a big enough change to warrant a new minor version.  I would also add something along the lines of "we'll no longer fall-back on granting role assignment contexts based on Enterprise membership when role assignment records have no explicit context (enterprise UUID or `*`)"